src/Common/Services/Mail/MailBootstrap.cs

```diff
@@ -1,39 +1,55 @@
+using System.Configuration;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Passwordless.Common.Services.Mail.Aws;
+using Passwordless.Common.Services.Mail.File;
+using Passwordless.Common.Services.Mail.SendGrid;
+using Passwordless.Common.Services.Mail.Smtp;
+
 namespace Passwordless.Common.Services.Mail;
 
 public static class MailBootstrap
 {
     public static void AddMail(this WebApplicationBuilder builder)
     {
-        builder.Services.AddOptions<MailConfiguration>().BindConfiguration("Mail");
-
-        if (builder.Configuration.GetSection("Mail:Postmark").Exists())
-        {
-            var configurationSection = builder.Configuration.GetSection("Mail:PostMark");
+        builder.Services.AddSingleton<IMailProviderFactory, MailProviderFactory>();
+        builder.Services.AddScoped<IMailProvider, OrderedMailProvider>();
+        builder.Services.Configure<MailConfiguration>(builder.Configuration.GetSection("Mail"))
+            .PostConfigure<MailConfiguration>(o =>
+            {
+                var section = builder.Configuration.GetSection("Mail:Providers");
 
-            var clients = new List<PostmarkClientConfiguration>();
-            configurationSection.GetSection("MessageStreams").Bind(clients);
+                // Add a default file provider if no providers are configured.
+                if (!section.GetChildren().Any())
+                {
+                    var fileProviderOptions = new FileMailProviderOptions();
+                    o.Providers.Add(fileProviderOptions);
+                    return;
+                }
```

Comment:
I think we should have a no-op mail provider in the default case. This may be useful in tests, where you don't care about emails at all and the app in the container may not have the permissions to have files.

Reply:
We can add support for it later in the integration testing project